### PR TITLE
Integration

### PR DESCRIPTION
**Pull request by 7imbrook**
### Merge checklist

What needs to be completed before Integration can be merged into master
- [x] Resolve merge conflicts
- [x] Complete everything in the milestone
- [x] Fully test Integration see more #99 
- [x] Fix any syntax issues
- [x] Remove any unnecessary log statements
- [x] Finish up the additional functionally of zooming and panning, see #65 #95 
- [x] Remove dead code

Pan and zoom functionality from the subview standpoint is complete, see [LectureViewChild](https://github.com/RITAccess/accessmath/blob/Integration/AccessLecture/LectureViewContainer.h#L19-L57) protocol. Things to keep in mind while implementing:
- Bounds don't change, they will aways be the full canvas size
- Frame gets resized with transforms, transforms can be accessed off the view but shouldn't be modified
- Two finger gestures should be avoided to not interfere with panning and zooming, also not to confuse the user
### Ready to merge?
- [x] @7imbrook
- [x] @piperchester 
- [x] @rasampratik

### CODE REVIEW

File: AccessLecture/ConnectionViewController.m [hunk 1]
```diff
@@ -40,6 +44,10 @@ - (void)viewDidLoad
     AccessLectureAppDelegate *app = [UIApplication sharedApplication].delegate;
     server = app.server;
     
+    _connectionAddress.text = @"michaeltimbrook.com";
```
Comment (7imbrook) [inline AccessLecture/ConnectionViewController.m]: Needs to be removed

File: AccessLecture/DrawViewController.m [hunk 2]
```diff
@@ -171,6 +171,9 @@ - (IBAction)undoButtonPress:(id)sender
 
 - (IBAction)shapeButtonPress:(id)sender
 {
+    
+    NSLog(@"%f, %f", self.drawView.bounds.size.height, self.drawView.bounds.size.width);
```
Comment (7imbrook) [inline AccessLecture/DrawViewController.m]: Log statement should be removed, the only log statements that can stay should be one that provide crash log information.

File: AccessLecture/FileManager.h [hunk 3]
```diff
@@ -45,7 +45,7 @@
 //  returns nil if no file was found
 //
 + (NSURL *)findFileIn:(NSArray *)files thatFits:(BOOL(^)(NSURL*))condition;
-
++(BOOL) saveDocument:(AccessDocument *)document;
```
Comment (7imbrook) [inline AccessLecture/FileManager.h]: Minor syntactical change `+ (BOOL)saveDoc..`
Space is after the -/+ and no space after the return type

File: AccessLecture/FileManager.m [hunk 4]
```diff
@@ -94,5 +95,46 @@ + (void) clearAllDocuments{
     }
     NSLog(@"Files succesfully deleted");
 }
++(BOOL) saveDocument:(AccessDocument *)document{
```
Comment (7imbrook) [inline AccessLecture/FileManager.m]: Syntax

File: AccessLecture/FileManager.m [hunk 5]
```diff
@@ -94,5 +95,46 @@ + (void) clearAllDocuments{
...
+    AccessDocument *currentDocument;
+    Lecture *currentLecture = document.lecture;
+    NSURL * currentDirectory = [FileManager iCloudDirectoryURL];
+    if (currentDirectory == nil) currentDirectory = [FileManager localDocumentsDirectoryURL];
+      
+  
+    NSString *docsPath =[[currentDirectory absoluteString] stringByAppendingString:[NSString stringWithFormat:@"AccessMath/%@.lecture",currentLecture.name]];
+    NSURL *docURL = [NSURL URLWithString:docsPath];
+    currentDocument = [[AccessDocument alloc] initWithFileURL:docURL];
+  //  currentLecture.image = UIImagePNGRepresentation(saveImage);
```
Comment (7imbrook) [inline AccessLecture/FileManager.m]: Remove commented lines if not used

File: AccessLecture/AccessLectureRuntime.m [hunk 6]
```diff
@@ -55,6 +55,7 @@ - (void)openDocument:(NSURL *) withURL{
             {
                 NSLog(@"Success");
```
Comment (7imbrook) [inline AccessLecture/AccessLectureRuntime.m]: Non descriptive log statement. Either expand it or remove.

File: AccessLecture/ConnectViewController.m [hunk 7]
```diff
@@ -34,11 +34,8 @@ - (void)connectToLecture:(NSString *)lecture {
...
     [app.server connect];
     [app.server getFullLecture:lecture completion:^(Lecture *lect, BOOL found) {
-        if (found) {
-            NSLog(@"%@", lect);
-        } else {
-            NSLog(@"Not Found");
-            [self showAlertViewWithMessage:[NSString stringWithFormat:@"%@ not found on server.", lecture]];
+        if (!found) {
+            NSLog(@"%@ Not Found", lect);
         }
```
Comment (7imbrook) [inline AccessLecture/ConnectViewController.m]: Remove I'm on it.

File: AccessLecture/ConnectionViewController.m [hunk 8]
```diff
@@ -9,15 +9,14 @@
 #import "ConnectionViewController.h"
 #import "AccessLectureAppDelegate.h"
 #import "ALNetworkInterface.h"
-
-@interface ConnectionViewController ()
-
-@end
+#import <QuartzCore/CATransform3D.h>
 
 @implementation ConnectionViewController {
```
Comment (7imbrook) [inline AccessLecture/ConnectionViewController.m]: Inline bracing should be used here

File: AccessLecture/ConnectionViewController.m [hunk 9]
```diff
@@ -79,55 +136,75 @@ - (IBAction)userDidCancel:(id)sender
...
-        [_previewView.layer addSublayer:preview];
+    return NO;
+}
+
+
+# pragma mark - Rotation Handling
+
+- (void)didRotateFromInterfaceOrientation:(UIInterfaceOrientation)fromInterfaceOrientation
+{
+    //  NOTE: decimal points in the CATransform are necessary.
```
Comment (7imbrook) [inline AccessLecture/ConnectionViewController.m]: We know, they're floats

File: AccessLecture/DrawViewController.m [hunk 10]
```diff
@@ -171,6 +171,7 @@ - (IBAction)undoButtonPress:(id)sender
 
 - (IBAction)shapeButtonPress:(id)sender
 {
+
```
Comment (7imbrook) [inline AccessLecture/DrawViewController.m:175]: No '\n' needed

File: AccessLecture/FTCoreTextView.m [hunk 11]
```diff
@@ -436,20 +436,20 @@ CTFontRef CTFontCreateFromUIFont(UIFont *font)
...
-			break;
-		case FTCoreTextAlignementRight:
-			return UITextAlignmentRight;
-			break;
-		default:
-			return UITextAlignmentLeft;
-			break;
-	}
-}
+//UITextAlignment UITextAlignmentFromCoreTextAlignment(FTCoreTextAlignement alignment)
```
Comment (7imbrook) [inline AccessLecture/FTCoreTextView.m]: Comment code should be removed

File: AccessLecture/FTCoreTextView.m [hunk 12]
```diff
@@ -436,20 +436,20 @@ CTFontRef CTFontCreateFromUIFont(UIFont *font)
...
+//			return UITextAlignmentRight;
+//			break;
+//		default:
+//			return UITextAlignmentLeft;
+//			break;
+//	}
+//}
 
 #pragma mark - FTCoreTextView business
 #pragma mark -
```
Comment (7imbrook) [inline AccessLecture/FTCoreTextView.m]: Multiple pragma marks should be removed

File: AccessLecture/LectureViewContainer.m [hunk 13]
```diff
@@ -10,22 +10,95 @@
...
+@implementation VCBlank
+
++ (instancetype)new
+{
+    VCBlank *new = [[VCBlank alloc] init];
+    [new.view setBackgroundColor:[UIColor whiteColor]];
+    return new;
+}
+- (UIView *)contentView { return self.view; }
+- (void)contentSizeChanging:(CGSize)size { }
```
Comment (7imbrook) [inline AccessLecture/LectureViewContainer.m]: Can be removed (Just line 53)

File: AccessLecture/NotesViewController.m [hunk 14]
```diff
@@ -7,13 +7,76 @@
 //
 
 #import "NotesViewController.h"
-#import "UILargeAlertView.h"
 
-@interface NotesViewController ()
+@interface noteLoader:NSObject{
```
Comment (7imbrook) [inline AccessLecture/NotesViewController.m]: Braces should be removed

File: AccessLecture/NotesViewController.m [hunk 15]
```diff
@@ -7,13 +7,76 @@
...
+    [aCoder encodeObject:_styleText forKey:TEXT_KEY];
+    [aCoder encodeObject:_noteType forKey:TYPE_KEY];
+    [aCoder encodeObject:_drawContent forKey:DRAW_KEY];
+}
 @end
+@interface NotesViewController ()
 
-@implementation NotesViewController
+@end
+@implementation NotesViewController{
```
Comment (7imbrook) [inline AccessLecture/NotesViewController.m]: Inline braces should be used

File: AccessLecture/ConnectViewController.m (hunk 7 above)
Comment (7imbrook) [inline AccessLecture/ConnectViewController.m]: Wait are we even using this class anymore?

File: AccessLecture/LectureViewContainer.m (hunk 13 above)
Comment (piperchester) [inline AccessLecture/LectureViewContainer.m]: I'll remove this.

**Comment by piperchester** [comment]
What more can we test? I'll take a look at DrawViewController.

**Comment by 7imbrook** [comment]
###### Ignoring any CI Build failures (I tried it again because XCTool was half working on my machine)

I am ready for this to merge after one last code review, see description for more.

File: AccessLecture Tests/DrawViewControllerTests.m [hunk 16]
```diff
@@ -0,0 +1,97 @@
...
+//
+
+#import <SenTestingKit/SenTestingKit.h>
+#import "DrawViewController.h"
+
+@interface DrawViewControllerTests : SenTestCase
+
+@end
+
+@implementation DrawViewControllerTests {
```
Comment (7imbrook) [inline AccessLecture Tests/DrawViewControllerTests.m]: Needs `\n` :end:

File: AccessLecture Tests/DrawViewTests.m [hunk 17]
```diff
@@ -0,0 +1,38 @@
...
+//
+
+#import <SenTestingKit/SenTestingKit.h>
+#import "DrawView.h"
+
+@interface DrawViewTests : SenTestCase
+
+@end
+
+@implementation DrawViewTests {
```
Comment (7imbrook) [inline AccessLecture Tests/DrawViewTests.m]: Needs a new line

File: AccessLecture/AccessLectureRuntime.m [hunk 18]
```diff
@@ -53,8 +57,7 @@ - (void)openDocument:(NSURL *) withURL{
         [_currentDocument openWithCompletionHandler:^(BOOL success){
             if(success)
             {
```
Comment (7imbrook) [inline AccessLecture/AccessLectureRuntime.m]: Ifs are Egyptian style with indents.

File: AccessLecture Tests/DrawViewControllerTests.m (hunk 16 above)
Comment (piperchester) [inline AccessLecture Tests/DrawViewControllerTests.m]: Wait, our Coding Guidelines say we're using Egyptian for class declarations. Why does this line need a `\n`?

File: AccessLecture Tests/DrawViewTests.m (hunk 17 above)
Comment (piperchester) [inline AccessLecture Tests/DrawViewTests.m]: Wait, our Coding Guidelines say we're using Egyptian for class declarations. Why does this line need a `\n`?

**Comment by piperchester** [comment]
Just pulled from Integration and had a conflict with the project file... was code signing changed? Or did Travis create the conflict?

File: AccessLecture Tests/DrawViewControllerTests.m (hunk 16 above)
Comment (7imbrook) [inline AccessLecture Tests/DrawViewControllerTests.m]: Whoops you're right! I feel like we should be using Inline for this, thoughts?

File: AccessLecture Tests/DrawViewTests.m (hunk 17 above)
Comment (7imbrook) [inline AccessLecture Tests/DrawViewTests.m]: See above...

File: AccessLecture Tests/DrawViewControllerTests.m (hunk 16 above)
Comment (piperchester) [inline AccessLecture Tests/DrawViewControllerTests.m]: That's what I proposed a couple weeks ago, but now we'd have to change all of our class declarations. Inline would look a lot more consistent in my opinion. Maybe we can start moving away from Egyptian for class declarations.

File: AccessLecture Tests/DrawViewTests.m (hunk 17 above)
Comment (piperchester) [inline AccessLecture Tests/DrawViewTests.m]: See above....again

File: AccessLecture/AccessLectureRuntime.m (hunk 18 above)
Comment (piperchester) [inline AccessLecture/AccessLectureRuntime.m]: I'll format this - :+1:

File: AccessLecture Tests/DrawViewControllerTests.m (hunk 16 above)
Comment (7imbrook) [inline AccessLecture Tests/DrawViewControllerTests.m]: Agreed, I updated the coding guidelines. I've been doing it inline for a while anyway thinking that was what we put down.

Comment (piperchester) [inline AccessLecture Tests/DrawViewControllerTests.m]: :ok_hand:

**Comment by 7imbrook** [comment]
@piperchester if you're getting conflicts pull and reset everything to FETCH_HEAD. Any of your local changes on Integration should have already been push or they're not getting merged.

**Comment by 7imbrook** [comment]
Alright I'm gonna finish up testing this by hand, and I'll merge in when ready (Then we're starting a new development branch)

**Comment by piperchester** [comment]
@7imbrook Did the new icons ever get integrated?

**Comment by 7imbrook** [comment]
You never merged them then, pull and get them in before 11 please and thank you

**Comment by piperchester** [comment]
Great work team! :tada:

**Pull request merged**